### ISSUE TITLE
Adapt disabled tests checking debug setlocal from PUC-Rio Lua 5.1 test suite in db.lua

### ISSUE THREAD

**Issue by Buristan** [tarantool/tarantool#5694]
This ticket is a logical continuation of gh-4473.

If run  PUC-Rio Lua 5.1 test suite by LuaJIT (HEAD at the moment: [`1672bdc`](https://github.com/tarantool/luajit/commit/1672bdc0ffbee9f32fadb6943909d6af7eb9a7b1)) (not Tarantool) it fails with the following error:
```
/home/burii/builds_workspace/luajit/gh-4473-adopt-luavela-test-suites/src/luajit: db.lua:189: assertion failed!
stack traceback:
        [C]: in function 'assert'
        db.lua:189: in function 'f'
        db.lua:218: in function 'g'
        db.lua:229: in function 'dofile'
        ...adopt-luavela-test-suites/test/PUC-Lua-5.1-tests/all.lua:82: in main chunk
        [C]: at 0x55688513bf96
```
caused by this test chunk (assuming that all previous failing test chunks non relating to this error are disabled):
```lua
function f(a,b)
  collectgarbage()
  local _, x = debug.getlocal(1, 1)
  local _, y = debug.getlocal(1, 2)
  assert(x == a and y == b)
  assert(debug.setlocal(2, 3, "pera") == "AA".."AA")
  assert(debug.setlocal(2, 4, "maçã") == "B")
  x = debug.getinfo(2)
  assert(x.func == g and x.what == "Lua" and x.name == 'g' and
         x.nups == 0 and string.find(x.source, "^@.*db%.lua"))
  glob = glob+1
  assert(debug.getinfo(1, "l").currentline == L+1)
  assert(debug.getinfo(1, "l").currentline == L+2)
end

-- <snipped>

function g(...)
  do local a,b,c; a=math.sin(40); end
  local feijao
  local AAAA,B = "xuxu", "mamão"
  f(AAAA,B)
  assert(AAAA == "pera" and B == "maçã")
  do
     local B = 13
     local x,y = debug.getlocal(1,5)
     assert(x == 'B' and y == 13)
  end
end

g()

```

`f()` function is called only from `g(...)` vararg function.
Lua interprets `...` in the vararg functions like additional first argument unlike LuaJIT does.

According to [this section](https://www.lua.org/manual/5.0/manual.html#2.5.8) in Lua 5.0 Reference Manual, Lua 5.0 collect all extra arguments in `arg` variable in the vararg functions.
> A vararg function does not adjust its argument list; instead, it collects all extra arguments into an implicit parameter, called arg. 

According to [this section](https://www.lua.org/manual/5.1/manual.html#7.1)  described changes in the language in Lua 5.1 Reference Manual:
> The vararg system changed from the pseudo-argument arg with a table with the extra arguments to the vararg expression. 

But this stack slot in vararg functions is still used by `...`.

**Comment by Buristan** [comment]
This test chunk can be adapted to LuaJIT behavior like this:

```diff
diff --git a/test/PUC-Lua-5.1-tests/db.lua b/test/PUC-Lua-5.1-tests/db.lua
index 0f22b67..96ce913 100644
--- a/test/PUC-Lua-5.1-tests/db.lua
+++ b/test/PUC-Lua-5.1-tests/db.lua
@@ -186,8 +186,9 @@ function f(a,b)
   -- Lua interprets ... in the vararg functions like additional
   -- first argument unlike LuaJIT does.
   -- See also https://github.com/tarantool/tarantool/issues/5694.
-  -- assert(debug.setlocal(2, 3, "pera") == "AA".."AA")
-  -- assert(debug.setlocal(2, 4, "maçã") == "B")
+  -- This test is adapted to LuaJIT behavior.
+  assert(debug.setlocal(2, 2, "pera") == "AA".."AA")
+  assert(debug.setlocal(2, 3, "maçã") == "B")
   x = debug.getinfo(2)
   assert(x.func == g and x.what == "Lua" and x.name == 'g' and
          x.nups == 0 and string.find(x.source, "^@.*db%.lua"))
@@ -216,13 +217,11 @@ function g(...)
   local feijao
   local AAAA,B = "xuxu", "mamão"
   f(AAAA,B)
-  -- Test is disabled for LuaJIT for now. See comment in f().
-  -- assert(AAAA == "pera" and B == "maçã")
+  assert(AAAA == "pera" and B == "maçã")
   do
      local B = 13
-     local x,y = debug.getlocal(1,5)
-     -- Test is disabled for LuaJIT for now. See comment in f().
-     -- assert(x == 'B' and y == 13)
+     local x,y = debug.getlocal(1,4)
+     assert(x == 'B' and y == 13)
   end
 end
 
```

**Comment by igormunkin** [comment]
The issue is resolved in scope of tarantool/luajit@e9d1f65 . The submodule was updated in tarantool in 2.8.0-249-g8a09e18ce (8a09e18), 2.7.1-210-g5f4298b6f (5f4298b), 2.6.2-203-g8343a8129 (8343a81), 1.10.9-119-g7a4c7d0b0 (7a4c7d0).
